src/benchq/data_structures/error_budget.py

```diff
@@ -22,6 +23,7 @@ class ErrorBudget:
     circuit_generation_weight: float = 1
     synthesis_weight: float = 1
     ec_weight: float = 1
+    synthesis_failure_tolerance_set_value: Optional[float] = None
```

Comment:
This is unpythonic and, more importantly, makes this structure flakey, because now the weights sometimes really correspond to weights, and sometimes they just don't. If we really need to sometimes override synthesis failure tolerance, we can do this as follows:

- Modify this structure so that it holds concrete values instead of weights
- Add static factory method to construct this budget from weights.
  - This assumes that weights are only secondary info and are not used in other code

So something along the lines of:

```python
@dataclass
class ErrorBudget:
    circuit_generation_failure_tolerance: float
    synthesis_failure_tolerance: float
    ec_failure_tolerance: float

    @staticmethod
    def from_weights(
        total_tolerance: float, 
        circuit_generation_weight: int, 
        ec_weight: int, 
        synthesis_weight: int
    ) -> ErrorBudget:
         # compute tolerances and return:
        return ErrorBudget(...)
```
And now we could easily swap for other values, e.g.:
```python
new_budget = replace(old_budget, synthesis_failure_tolerance=1e-2)
```

If you don't think the sketch above solves what you were trying to achieve, then let me know and we can think of another solution.

Reply:
This seems good! It's a lot more flexible than what we have now.